packages/augur-sdk/src/state/getter/Platform.ts

```diff
@@ -0,0 +1,244 @@
+import * as t from 'io-ts';
+import { BigNumber } from 'bignumber.js';
+import { DB } from '../db/DB';
+import { Getter } from './Router';
+import { Augur } from '../../index';
+import * as _ from 'lodash';
+import {
+  Address,
+  CompleteSetsPurchasedLog,
+  CompleteSetsSoldLog,
+  DisputeCrowdsourcerContributionLog,
+  DisputeCrowdsourcerRedeemedLog,
+  InitialReporterRedeemedLog,
+  InitialReportSubmittedLog,
+  Log,
+  MarketCreatedLog,
+  ParsedOrderEventLog,
+  ParticipationTokensRedeemedLog,
+  Timestamped,
+  TradingProceedsClaimedLog
+} from "../logs/types";
+import { NULL_ADDRESS } from "./types";
+
+export interface PlatformActivityStatsResult {
+  activeUsers: number;
+
+  // OrderEvent table for fill events (eventType == 3) where they are the orderCreator or orderFiller address
+  // if multiple fills in the same tx count as one trade then also counting just the unique tradeGroupId from those
+  numberOfTrades: number;
+  openInterest: BigNumber;
+
+  // MarketCreated logs
+  marketsCreated: number;
+
+  volume: BigNumber;
+  amountStaked: BigNumber;
+  disputedMarkets: number;
+}
+
+export class Platform {
+  static getPlatformActivityStatsParams = t.intersection([
+    t.type({
+      universe: t.string,
+    }),
+    t.partial({
+      endTime: t.number,
+      startTime: t.number,
+    })]);
+
+  @Getter('getPlatformActivityStatsParams')
+  static async getPlatformActivityStats(
+    augur: Augur,
+    db: DB,
+    params: t.TypeOf<typeof Platform.getPlatformActivityStatsParams>
+  ): Promise<PlatformActivityStatsResult> {
+    const { universe } = params;
+    const startTime = params.startTime || 0;
+    const endTime = params.endTime || (await augur.contracts.augur.getTimestamp_()).toNumber();
+
+    if (Number(startTime) > Number(endTime)) {
+      throw new Error('startTime must be less than or equal to endTime');
+    }
+
+    return {
+      activeUsers: await getActiveUsers(universe, startTime, endTime, db),
+      numberOfTrades: await getTradeCount(universe, startTime, endTime, db),
+      openInterest: await getOpenInterest(universe, augur),
+      marketsCreated: await getMarketCount(universe, startTime, endTime, db),
+      volume: await getVolume(universe, startTime, endTime, db),
+      amountStaked: await getAmountStaked(universe, startTime, endTime, db),
+      disputedMarkets: await getDisputedMarkets(universe, startTime, endTime, db),
+    };
+  }
+}
+
+async function getActiveUsers(
+  universe: string,
+  startTime: number,
+  endTime: number,
+  db: DB
+): Promise<number> {
+  const getField = makeGetField(universe, startTime, endTime);
+  const users: Address[] = [
+    ...await getField<CompleteSetsPurchasedLog>('account', db.findCompleteSetsPurchasedLogs.bind(db)),
```

Comment:
You *could* change the function declaration in db vs calling bind on these. If we are going to build many of these *lense* functions that might be the way to go.